test/service/test_manifest.py

```diff
@@ -784,6 +797,127 @@ def test_full_metadata_missing_fields(self):
+                        file_names = defaultdict(list)
+
+                        # Run the generation of manifests twice to verify generated file names are the same when re-run
+                        for project_id in project_ids * 2:
+                            params = {'filters': json.dumps({'projectId': {'is': [project_id]}}), 'format': 'full'}
+                            response = self.get_manifest(params)
+                            self.assertEqual(200, response.status_code, 'Unable to download manifest')
+                            file_name = urlparse(response.url).path
+                            file_names[project_id].append(file_name)
+
+                        self.assertEqual(file_names.keys(), set(project_ids))
+                        self.assertEqual([2, 2], list(map(len, file_names.values())))
+                        self.assertEqual([1, 1], list(map(len, map(set, file_names.values()))))
+
+    def test_get_seconds_until_expire(self):
+        """
+        Verify a header with valid x-amz-expiration and LastModified values returns the correct expiration value
+        """
+        with mock.patch('time.time', new=lambda: 1547691253.07010):
+            dt_now = datetime.now(timezone.utc)
+            for hours_last_modified in range(0, 25, 6):
+                with self.subTest(hours_last_modified=hours_last_modified):
+                    hours_till_expiration = (config.manifest_expiration * 24) - hours_last_modified
+                    expire_date = (dt_now + timedelta(hours=hours_till_expiration)).strftime("%a, %d %b %Y %H:%M:%S %Z")
+                    header = {
+                        'x-amz-expiration': f'expiry-date="{expire_date}", rule-id="Test Rule"',
+                        'LastModified': dt_now - timedelta(hours=hours_last_modified)
+                    }
+                    seconds_remaining = ManifestResponse._get_seconds_until_expire(header)
+                    self.assertAlmostEqual(seconds_remaining, 60 * 60 * hours_till_expiration, delta=1)
+
+    def test_get_seconds_until_expire_logs(self):
+        """
+        Verify a header with mismatched x-amz-expiration and LastModified values produces an error log message
+        """
+        with mock.patch('time.time', new=lambda: 1547691253.07010):
+            dt_now = datetime.now(timezone.utc)
+
+            def log_messages_from_get_seconds(expire_date: datetime, last_modified: datetime) -> List[str]:
+                expire_date_string = expire_date.strftime("%a, %d %b %Y %H:%M:%S %Z")
+                header = {
+                    'x-amz-expiration': f'expiry-date="{expire_date_string}", rule-id="Test Rule"',
+                    'LastModified': last_modified
+                }
+                with self.assertLogs(level='ERROR') as logs:
+                    ManifestResponse._get_seconds_until_expire(header)
+                    logger.error('Dummy error message to have at least one error message logged')
```

Comment:
What would happen if this line didn't exist?

Reply:
As part of the test case I verify that a log message is added when the two date fields (expire date & last modified) don't add up correctly, and that no log message is added when they do. Without a dummy log message, `assertLogs` throws an exception when no message is logged.
I could add a comment or just make the dummy log message more explanatory
```
logger.error('Dummy log message so assertLogs() does not fail if no other error log is generated')
```